pkg/controller/storage/snapshotclass/snapshotclass_controller.go

```diff
@@ -0,0 +1,216 @@
+/*
+
+ Copyright 2021 The KubeSphere Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+*/
+
+package snapshotclass
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	storagev1 "k8s.io/api/storage/v1"
+
+	snapshotv1beta1 "github.com/kubernetes-csi/external-snapshotter/client/v3/apis/volumesnapshot/v1beta1"
+	snapshotclient "github.com/kubernetes-csi/external-snapshotter/client/v3/clientset/versioned/typed/volumesnapshot/v1beta1"
+	snapinformers "github.com/kubernetes-csi/external-snapshotter/client/v3/informers/externalversions/volumesnapshot/v1beta1"
+	snapshotlisters "github.com/kubernetes-csi/external-snapshotter/client/v3/listers/volumesnapshot/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	storageinformersv1 "k8s.io/client-go/informers/storage/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	storagelistersv1 "k8s.io/client-go/listers/storage/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	crdscheme "kubesphere.io/kubesphere/pkg/client/clientset/versioned/scheme"
+)
+
+const annotationAllowSnapshot = "storageclass.kubesphere.io/allow-snapshot"
+
+type VolumeSnapshotClassController struct {
+	storageClassLister  storagelistersv1.StorageClassLister
+	storageClassSynced  cache.InformerSynced
+	snapshotClassClient snapshotclient.VolumeSnapshotClassInterface
+	snapshotClassLister snapshotlisters.VolumeSnapshotClassLister
+	snapshotClassSynced cache.InformerSynced
+
+	snapshotClassWorkQueue workqueue.RateLimitingInterface
+}
+
+//This controller is responseible to watch StorageClass
```

Comment:
responseible => responsible, and should leave spaces between `//` and real comment, the rest of this PR is OK

Reply:
Okay, I have updated it.